C_CPP: Overloading Unary Operators Using 'friend' Functions (#7127)
girlscript/winter-of-contributing: C_CPP/OOPs in C++/Operator Overloading/overloading-unary-operator.md

# Overloading Unary Operators Using 'friend' Functions

> Here, the operator functions is not the member function of any class. Hence, in this scenario if an operator has to be overloaded to make it work on a class object, the operator function has to be declared as a **friend** to that class.  

Reviewer: ```suggestion
Here, the operator function is not the member function of any class. Hence, in this scenario if an operator has to be overloaded to make it work on a class object, the operator function has to be declared as a **friend** to that class.  
```

In other words, when operator function is *friend* to a class, then we can call that function as a "friend operation function."  

> The synatx for defining a friend operator function to overload an unary operator is as follows:  

Reviewer: ```suggestion
The syntax for defining a friend operator function to overload a unary operator is as follows:  
```

```C++
class < class_name >
{
   public:
         friend < return_type > operator opr (< argument >);
};
< return_type > operator opr (< argument >)
{
   //function definition
}
```  
where  

> *friend* -> keyword to make a non-member function as a friend to a class  
> *return_type* -> type of the value returned by the friend operator function  
> *operator* -> keyword to create an operator function  
> *opr* -> unary operator that is to be overloaded  
> *argument* -> is the formal argument referring to class object on which the operator overloading is being done.  

Reviewer: ```suggestion
- *friend* -> keyword to make a non-member function as a friend to a class  
- *return_type* -> type of the value returned by the friend operator function  
- *operator* -> keyword to create an operator function  
- *opr* -> unary operator that is to be overloaded  
- *argument* -> is the formal argument referring to class object on which the operator overloading is being done.  
```

do some markdown formatting like this.

> *argument* -> is the formal argument referring to class object on which the operator overloading is being done.  
> 
> Here's a program to overload an unary minus '-' operator using friend functions.  
> 

Reviewer: ```suggestion
*argument* -> is the formal argument referring to the class object on which the operator overloading is being done.  

Here's a program to overload a unary minus '-' operator using friend functions.  

```
